Mathlib/RepresentationTheory/Homological/GroupHomology/LowDegree.lean

```diff
@@ -430,4 +436,193 @@ lemma single_one_snd_sub_single_one_snd_mem_boundaries₂ (g h : G) (a : A) :
   simp [d₃₂]
 
 end Boundaries
+
+section IsCycle
+
+section
+
+variable {G A : Type*} [Mul G] [Inv G] [AddCommGroup A] [SMul G A]
+
+/-- A finsupp `∑ aᵢ·gᵢ : G →₀ A` satisfies the 1-cycle condition if `∑ gᵢ⁻¹ • aᵢ = ∑ aᵢ`. -/
+def IsCycle₁ (x : G →₀ A) : Prop := x.sum (fun g a => g⁻¹ • a) = x.sum (fun _ a => a)
+
+/-- A finsupp `∑ aᵢ·(gᵢ, hᵢ) : G × G →₀ A` satisfies the 2-cycle condition if
+`∑ (gᵢ⁻¹ • aᵢ)·hᵢ + aᵢ·gᵢ = ∑ aᵢ·gᵢhᵢ`. -/
+def IsCycle₂ (x : G × G →₀ A) : Prop :=
+  x.sum (fun g a => single g.2 (g.1⁻¹ • a) + single g.1 a) =
+    x.sum (fun g a => single (g.1 * g.2) a)
+
+end
+
+section
+
+variable {G A : Type*} [Group G] [AddCommGroup A] [DistribMulAction G A]
+
+@[simp]
+theorem single_isCycle₁_iff (g : G) (a : A) :
+    IsCycle₁ (single g a) ↔ g • a = a := by
+  rw [← (MulAction.bijective g⁻¹).1.eq_iff]
+  simp [IsCycle₁, eq_comm]
+
+theorem single_isCycle₁_of_mem_fixedPoints
+    (g : G) (a : A) (ha : a ∈ MulAction.fixedPoints G A) :
+    IsCycle₁ (single g a) := by
+  simp_all [IsCycle₁]
+
+theorem single_isCycle₂_iff_inv (g : G × G) (a : A) :
+    IsCycle₂ (single g a) ↔
+      single g.2 (g.1⁻¹ • a) + single g.1 a = single (g.1 * g.2) a := by
+  simp [IsCycle₂]
+
+@[simp]
+theorem single_isCycle₂_iff (g : G × G) (a : A) :
+    IsCycle₂ (single g a) ↔
+      single g.2 a + single g.1 (g.1 • a) = single (g.1 * g.2) (g.1 • a) := by
+  rw [← (Finsupp.mapRange_injective (α := G) _ (smul_zero _) (MulAction.bijective g.1⁻¹).1).eq_iff]
+  simp [mapRange_add, IsCycle₂]
+
+end
+
+end IsCycle
+
+section IsBoundary
+
+section
+
+variable {G A : Type*} [Mul G] [Inv G] [AddCommGroup A] [SMul G A]
+
+variable (G) in
+/-- A term `x : A` satisfies the 0-boundary condition if there exists a finsupp
+`∑ aᵢ·gᵢ : G →₀ A` such that `∑ gᵢ⁻¹ • aᵢ - aᵢ = x`. -/
+def IsBoundary₀ (a : A) : Prop :=
+  ∃ (x : G →₀ A), x.sum (fun g a => g⁻¹ • a - a) = a
```

Comment:
I think it's slightly confusing to have `a` as both a free and bound variable here.

Reply:
done, thanks